cola/diffparse.py

```diff
@@ -155,7 +155,7 @@ def parse(self, diff_text):
                     new.parse(parts[3][1:])
                     lines.append((self.DASH, self.DASH, self.DASH))
                     continue
-            if state == INITIAL_STATE:
+            if state == INITIAL_STATE or text == '\\ No newline at end of file':
```

Comment:
Move the `\\ No newline ...` string contents into a "constant" variable called `NO_NEWLINE`, which would be consistent with another similar usage in this file.  That'll make this line a little easier to read as `if state == INITIAL_STATE or text == NO_NEWLINE:`